matchms/networking/SimilarityNetwork.py

```diff
@@ -0,0 +1,135 @@
+from matchms import Scores
+from .networking_functions import get_top_hits
+
+
+class SimilarityNetwork:
+    """Create a spectal network from spectrum similarities.
+
+    For example
+
+    .. testcode::
+
+        import numpy as np
+        from matchms import Spectrum, calculate_scores
+        from matchms.similarity import ModifiedCosine
+        from matchms.networking import SimilarityNetwork
+
+        spectrum_1 = Spectrum(mz=np.array([100, 150, 200.]),
+                              intensities=np.array([0.7, 0.2, 0.1]),
+                              metadata={"precursor_mz": 100.0,
+                                        "testID": "one"})
+        spectrum_2 = Spectrum(mz=np.array([104.9, 140, 190.]),
+                              intensities=np.array([0.4, 0.2, 0.1]),
+                              metadata={"precursor_mz": 105.0,
+                                        "testID": "two"})
+
+        # Use factory to construct a similarity function
+        modified_cosine = ModifiedCosine(tolerance=0.2)
+        spectrums = [spectrum_1, spectrum_2]
+        scores = calculate_scores(spectrums, spectrums, modified_cosine)
+        ms_network = SimilarityNetwork(identifier="testID")
+        ms_network.create_network(scores)
+
+        print("Nodes:", ms_network.graph.nodes())
+
+    Should output
+
+    .. testoutput::
+
+        Nodes: ['one', 'two']
+
+    """
+    def __init__(self, identifier: str = "spectrumid",
+                 top_n: int = 20,
+                 max_links: int = 10,
+                 score_cutoff: float = 0.7,
+                 link_method: str = 'single'):
+        """
+        Parameters
+        ----------
+        identifier
+            Unique intentifier for each spectrum in scores. Will also be used for
+            node names.
+        top_n
+            Consider edge between spectrumA and spectrumB if score falls into
+            top_n for spectrumA or spectrumB (link_method="single"), or into
+            top_n for spectrumA and spectrumB (link_method="mutual"). From those
+            potential links, only max_links will be kept, so top_n must be >= max_links.
+        max_links
+            Maximum number of links to add per node. Default = 10.
+            Due to incoming links, total number of links per node can be higher.
+        score_cutoff
+            Threshold for given similarities. Edges/Links will only be made for
+            similarities > score_cutoff. Default = 0.7.
+        link_method
+            Chose between 'single' and 'mutual'. 'single will add all links based
+            on individual nodes. 'mutual' will only add links if that link appears
+            in the given top-n list for both nodes.
+        """
+        # pylint: disable=too-many-arguments
+        self.identifier = identifier
+        self.top_n = top_n
+        self.max_links = max_links
+        self.score_cutoff = score_cutoff
+        self.link_method = link_method
+        self.graph = None
```

Comment:
Can you document what self.graph is? For example
```suggestion
        self.graph: Optional[nx.Graph] = None
        """NetworkX graph. Set after calling create_network()"""
```